contracts/yellow/YellowClearingBase.sol

```diff
@@ -0,0 +1,207 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import '@openzeppelin/contracts/access/AccessControl.sol';
+import '@openzeppelin/contracts/utils/cryptography/ECDSA.sol';
+import './IVault.sol';
+
+abstract contract YellowClearingBase is AccessControl {
+    using ECDSA for bytes32;
+
+    // Participant status
+    enum ParticipantStatus {
+        // Participant is not registered
+        None,
+        // Participant is registered but not active
+        Pending,
+        // Participant is registered but do not have token staked
+        Inactive,
+        // Participant is registered and have token staked
+        Active,
+        // Participant is registered but is not allowed to participate
+        Suspended,
+        // Participant is registered but have migrated to the next implementation
+        Migrated
+    }
+
+    // Participant data
+    struct ParticipantData {
+        ParticipantStatus status;
+        IVault vault;
+        uint256 registrationBlock;
+    }
```

Comment:
Re-order fields like this:
- registrationTime (since it will be replaced with timestamp instead of block number)
- status
- vault

Reply:
Bigger to less prioritized?

Reply:
8f6b4f1b8c54a7035f5c81ffadc2bb390d58a88c